File: client/src/mutations/update_article_title.js
import gql from 'graphql-tag';

const updateArticleTitle = gql`
    mutation updateArticleTitle($id: Int!, $title: String!) {
        updateArticleTitle(id: $id, title: $title) {
            id
            body
            title
        }
    }
`;

export default updateArticleTitle;

File: app/graphql/mutations/update_article_title.rb
module Mutations
    class UpdateArticleTitle < BaseMutation
      argument :id, Int, required: true
      argument :title, String, required: true
      
      type Types::ArticleType
  
      def resolve(id: nil, title: "", body: "")
          article = Article.find(id)
          return unless article
          #Return if article isn't found, add error message later.
          article.title = title
        if  article.save
            SwyleSchema.subscriptions.trigger("articleUpdated", {}, article)
            article
        else
          return article.errors.full_messages
        end  
      end
    end
  end